tests/src/test/java/org/jboss/forge/furnace/container/guice/service/AddonRegistryTest.java

```diff
@@ -57,4 +61,18 @@ public void testServiceInjection() throws Exception
       Assert.assertThat(service.isUnsatisfied(), is(false));
       Assert.assertThat(service.get(), notNullValue());
    }
+
+   @Test
+   public void testExportedTypes() throws Exception
```

Comment:
Maybe we should create some abstract test for the containers to verify that they expose only what is supposed to be exported? Not sure if it's doable because of the semantics of each container, but at least some smoke testing would be good I think